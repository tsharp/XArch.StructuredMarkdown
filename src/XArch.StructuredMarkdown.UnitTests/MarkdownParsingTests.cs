using System.Text;

using Snapshooter.Xunit;

using XArch.StructuredMarkdown.Parsing;
using XArch.StructuredMarkdown.Serialization;

namespace XArch.StructuredMarkdown.UnitTests
{
    public class MarkdownParsingTests
    {
        [Fact]
        public void CanParseOnlyFrontmatter()
        {
            string markdown = File.ReadAllText("./test-data/markdown/only_frontmatter.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            document.MatchSnapshot();
        }

        [Fact]
        public void CanParseFullDocumentWithNoFrontMatter()
        {
            string markdown = File.ReadAllText("./test-data/markdown/structured_markdown_no_frontmatter.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            document.MatchSnapshot();
        }

        [Fact]
        public void CanParseAndSerializeFullDocument2()
        {
            string markdown = File.ReadAllText("./test-data/markdown/structured_markdown_with_frontmatter.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            StructuredMarkdownSerializer
                .ToMarkdown(document)
                .MatchSnapshot();
        }

        [Fact]
        public void CanParseAndSerializeFullDocument()
        {
            string markdown = File.ReadAllText("./test-data/markdown/structured_markdown_no_frontmatter.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            StructuredMarkdownSerializer
                .ToMarkdown(document)
                .MatchSnapshot();
        }

        [Fact]
        public void CanParseSection()
        {
            string markdown = File.ReadAllText("./test-data/markdown/section.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            document.MatchSnapshot();
        }

        [Fact]
        public void CanParseParagraph()
        {
            string markdown = File.ReadAllText("./test-data/markdown/paragraph.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            document.MatchSnapshot();
        }
    }
}
