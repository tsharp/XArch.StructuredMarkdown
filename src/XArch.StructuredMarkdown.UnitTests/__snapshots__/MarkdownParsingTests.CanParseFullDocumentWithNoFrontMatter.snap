{
  "Root": {
    "Title": null,
    "Parts": [
      {
        "Title": null,
        "Parts": [
          {
            "Content": "I have content to go here, but I don't want to use frontmatter. This is a test document to check how the parser handles structured markdown without frontmatter.",
            "Metadata": null
          },
          {
            "Title": null,
            "Parts": [
              {
                "Content": "I have content to go here, but I don't want to use frontmatter. This is a test document to check how the parser handles structured markdown without frontmatter.",
                "Metadata": null
              }
            ],
            "Metadata": {
              "section": "nested-sub-section",
              "tags": [
                "external-section"
              ]
            }
          }
        ],
        "Metadata": {
          "section": "named-sub-section",
          "tags": [
            "external-section"
          ]
        }
      },
      {
        "Title": "Structured Markdown Test Document",
        "Parts": [
          {
            "Content": "This document is designed to test the functionality of the Structured Markdown parser and renderer. It includes various elements such as headings, lists, code blocks, and links.",
            "Metadata": null
          },
          {
            "Content": "This is a paragraph that introduces the document. It provides an overview of what to expect in the following sections.",
            "Metadata": {
              "tags": [
                "paragraph"
              ],
              "content_level": "hidden"
            }
          },
          {
            "Title": "Headings",
            "Parts": [
              {
                "Content": "This is a section about headings. It includes different levels of headings to test the rendering.",
                "Metadata": null
              },
              {
                "Title": "Heading Level 3",
                "Parts": [
                  {
                    "Content": "This is a level 3 heading. It should be rendered as a smaller heading than level 2.",
                    "Metadata": null
                  }
                ],
                "Metadata": {
                  "tags": [
                    "heading-3"
                  ],
                  "content_level": "paid_premium"
                }
              }
            ],
            "Metadata": {
              "tags": [
                "heading-2"
              ],
              "content_level": "paid_basic"
            }
          }
        ],
        "Metadata": {
          "tags": [
            "heading-1"
          ],
          "content_level": "basic"
        }
      }
    ],
    "Metadata": null
  },
  "Metadata": null
}
